packages/app/src/cli/models/app/app.ts

```diff
@@ -250,12 +261,16 @@ export class App implements AppInterface {
       (extension) => extension.type === specification.identifier || extension.type === specification.externalIdentifier,
     )
   }
+
+  specificationForIdentifier(identifier: string) {
+    return this.specifications.find((specification) => specification.identifier === identifier)
+  }
```

Comment:
We already have a `findSpecificationForType` on `loader.ts` (line 132). We also compare the `externalIdentifier` and the `additionalIdentifiers` (used by functions)
Do you think we can reuse that somehow or do we need to keep both?

Reply:
Should this one also compare the "externalIdentifier"?

Reply:
Do you mean like this Isaac? `this.specifications.find((specification) => specification.identifier === identifier || specification.externalIdentifier )`

Reply:
yeah, this is because for legacy reasons, some extensions were scaffolded with the "externalIdentifier". So better to always check both.
Also for functions, again for legacy reasons, the identifier can be "function" or any of the "additionalIdentifiers" that we've been using before unified config.

Reply:
> We already have a findSpecificationForType on loader.ts (line 132). We also compare the externalIdentifier and the additionalIdentifiers (used by functions)
Do you think we can reuse that somehow or do we need to keep both?

I couldn't figure out how to access the loader inside `release` and `prompts` - do you mind if we fix this in a follow up PR?

Reply:
yep, no worries

Reply:
Going back to this, the `findSpecificationForType` function is now a public function, can you call that one which is more complete either from here or from wherever you are calling `specificationForIdentifier`?

Reply:
✅ The `specificationForIdentifier` function has been updated to call `findSpecificationForType` in 6db37ad5e7eec6c1032e0a44eeafbb82d68ec3d9.